src/Bicep.Decompiler/DecompileOptions.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Decompiler;
+
+public record DecompileOptions
```

Comment:
[nit] To take advantage of the [record capabilities](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record) - immutability, value comparisons, deconstructors & the `with` syntax, I believe you'll need the following (defines the constructor, plus 2 public immutable properties):
```csharp
public record DecompileOptions(
    AllowMissingParamsAndVars = false,
    AllowMissingParamsAndVarsInNestedTemplates = false);
```

Here it looks like this'll behave like a regular `class` with public mutable fields.